journals/apps/journals/blocks.py

```diff
@@ -83,6 +87,62 @@ class JournalRichTextBlock(blocks.RichTextBlock):
     def get_searchable_content(self, value):
         return [parser(value.source, 'html.parser').get_text(' ')]
 
+    def expand_db_html(self, html, for_editor=False, base_url='/'):
+        """
+        Override from wagtail.wagtailcore.rich_text to use full path
+        to embedded images
+        """
+        def replace_a_tag(m):
+            """
+            overridden, return href for Pages and Documents that
+            journals frontend knows how to display
+            """
+            attrs = extract_attrs(m.group(1))
+            if 'linktype' not in attrs:
+                # return unchanged
+                return m.group(0)
+            if attrs['linktype'] == 'page':
+                try:
+                    page = Page.objects.get(id=attrs['id'])
+                    return '<a href="{page_path}">'.format(page_path=page.specific.get_frontend_page_path())
```

Comment:
NOTE: I don't think there's anything (simple) we can do about it, but I just want to note that this will cause a full page reload in frontend when the user clicks on it. If we're seeing this used a lot once live, we may want to follow up with a smoother option.

Reply:
noted